docs/get_started/quickstart.mdx

```diff
@@ -39,122 +37,299 @@ Get all the dependencies and build tools:
-</TabItem>
-<TabItem value="java" label="Java" default>
+</Step>
+<Step stepLabel="2" title="Build and run the service">
 
-Grab the Java Hello world example via the Restate CLI:
+Now, you are all set to start developing your service in `src/app.ts`.
+You can run the app with `ts-node-dev` while developing:
 
 ```shell
-restate example java-hello-world-gradle && cd java-hello-world-gradle
+npm run app-dev
 ```
 
-Or via `wget`:
+Once you have implemented your service, build the app and run it:
+
 ```shell
-wget https://github.com/restatedev/examples/releases/latest/download/java-hello-world-gradle.zip && unzip java-hello-world-gradle.zip -d java-hello-world-gradle && rm java-hello-world-gradle.zip && cd java-hello-world-gradle
+npm run build
+npm run app
 ```
 
-</TabItem>
-</Tabs>
+This starts the greeter service on port `9080`.
 
-## Step 2: Build and run the service
+</Step>
 
-<Tabs groupId="sdk" queryString>
-<TabItem value="ts" label="TypeScript" default>
+<Step stepLabel="3" title="Launch Restate">
+<CH.Code rows={"2"}>
 
-Now, you are all set to start developing your service in `src/app.ts`.
-You can run the app with `ts-node-dev` while developing:
+```shell npx
+npx @restatedev/restate-server
+```
 
-```shell
-npm run app-dev
+``` shell Homebrew
+brew install restatedev/tap/restate-server && restate-server
 ```
 
-Once you have implemented your service, build the app and run it:
+```shell Docker
+docker run --name restate_dev --rm -p 8080:8080 -p 9070:9070 -p 9071:9071 \
+    --add-host=host.docker.internal:host-gateway docker.io/restatedev/restate:VAR::RESTATE_VERSION
+```
+
+</CH.Code>
+
+<Admonition type="tip" icon="💡" title="Single binary">
+Restate is a single self-contained binary. No external dependencies needed. Check out our [download page](https://restate.dev/get-restate/) for other ways to run Restate.
+</Admonition>
+
+</Step>
+
+<Step stepLabel="4" title="Register the service">
+
+Now, we need to tell Restate where the service is running.
+You can use the Restate CLI via `npx` or via a global `npm` install.
+Alternatively, you can use `curl`:
+
+<CH.Code rows={"2"}>
+
+```shell CLI
+restate dp add http://localhost:9080
+```
+
+```shell curl
+curl localhost:9070/deployments -H 'content-type: application/json' \
+    -d '{"uri": "http://localhost:9080"}'
+```
+
+</CH.Code>
+
+If you run Restate with Docker, you need to use `http://host.docker.internal:9080` instead of `http://localhost:9080` to reach the service deployment.
+
+Restate then sends a request to the service deployment to ask which services and handlers
+are running behind this endpoint and will remember these.
+After executing this curl request, you should see the registered services and handlers printed to your terminal.
+
+</Step>
+
+<Step stepLabel="5" title="Send requests to the service">
+
+Invoke the function via:
 
 ```shell
-npm run build
-npm run app
+curl localhost:8080/myservice/hello -H 'content-type: application/json' \
+    -d '{"name": "Restate"}'
```

Comment:
This looks incorrect? can it be the input just the json string, like for java/kotlin? Also isn't the template creating a virtual object?

Perhaps we need to update this template.

Reply:
Indeed, I mentioned in the PR description that we need to give the TS quickstart another pass after we update the TS template ;)